### DISCUSSION TITLE
kernel can't mount ext4

### DISCUSSION THREAD

**Discussion by archandanime** [gtxaspec/wz_mini_hacks#159] [Development]
I compiled a kernel and flashed to kernel partition.
`kernel.config` has
```
CONFIG_EXT4_FS=y
CONFIG_JBD2=y
```
however, when I try to mount ext4-formatted SD card from initramfs debugging shell
```
/ # mount -t ext4 -o ro /dev/mmcblk0p1 /sdcard
[  207.230877] JBD2: Unrecognised features on journal
[  207.235876] EXT4-fs (mmcblk0p1): error loading journal
mount: mounting /dev/mmcblk0p1 on /sdcard failed: Invalid argument
```
an empty ext4 SD card can be mounted but after I wrote data from my PC, it can't mount anymore. Disabling journal by `sudo tune2fs -O ^has_journal /dev/sdd1` doesn't solve the issue.

**Comment by gtxaspec** [comment]
hmm... maybe you need to set mdev

`echo /sbin/mdev > /proc/sys/kernel/hotplug`

**Comment by archandanime** [comment]
thank you but... I think it's the journal features that the kernel can't understand thus can't mount. I could be wrong
the un-mountable ext4 has `journal_64bit` `journal_checksum_v3` features but the mountable doesn't
I thought supporting ext4 means supporting these features
Edit: it's because `journal_checksum_v3` is not supported in old kernels, the only way to make journal work is to backport its features

**Comment by archandanime** [comment]
creating ext4 partition without journal using `mkfs.ext4 -O ^has_journal /dev/sdd1` solves the problem but... the partition is more likely to be corrupted
